tests/test_BT_piecewise.py

```diff
@@ -0,0 +1,560 @@
+            are_toas_within_group_boundaries_mjd_method_per_parameter[0][0]
+            != are_toas_within_group_boundaries_mjd_method_per_parameter[1][0],
+        )
+
+        is_toa_in_each_group = [is_toa_in_each_group[0], is_toa_in_each_group[1]]
+
+        are_toas_within_group_boundaries_mjd_method_per_parameter = [
+            where_toas_should_be_in_group_1,
+            where_toas_should_be_in_group_2,
+        ]
+
+        does_in_piece_and_mjd_method_agree = np.array_equal(
+            are_toas_within_group_boundaries_mjd_method_per_parameter,
+            is_toa_in_each_group,
+        )
+
+        assert does_in_piece_and_mjd_method_agree
+
+    elif param == "non-complete group coverage":
+        # This test is performed to make sure toas that shouldn't be in any group are correctly being flagged. Only later half of toas should be in group. Distinct from the overlapping group test since its not ambiguous which group they belong to since they aren't in a group
+
+        boundaries = m_piecewise.get_group_boundaries()
+
+        lower_edge_of_group = boundaries[0][0]
+        are_toas_above_group_edge = [barycentric_toa.value >= lower_edge_of_group]
+
+        do_in_piece_and_mjd_methods_of_assigning_groups_agree = np.array_equal(
+            is_toa_in_each_group, are_toas_above_group_edge
+        )
+
+        assert do_in_piece_and_mjd_methods_of_assigning_groups_agree
+
+
+def add_offset_in_model_parameter(indexes, param, model):
+    m_piecewise_temp = deepcopy(model)
+    parameter_string = f"{param}_{convert_int_into_index(indexes)}"
+    if hasattr(m_piecewise_temp, parameter_string):
+        delta = getattr(m_piecewise_temp, parameter_string).value + 1e-5
+        getattr(m_piecewise_temp, parameter_string).value = delta
+        m_piecewise_temp.setup()
+    else:
+        parameter_string = param[0:2]
+        getattr(m_piecewise_temp, parameter_string).value = (
+            getattr(m_piecewise_temp, parameter_string).value + 1e-5
+        )
+        m_piecewise_temp.setup()
+    return m_piecewise_temp
+
+
+@pytest.mark.parametrize("param, index", [("T0X", 0), ("T0X", 1)])
+def test_residuals_in_groups_respond_to_changes_in_corresponding_piecewise_parameter(
+    build_model, build_piecewise_model_with_two_pieces, param, index
+):
+    m_piecewise, toas = add_groups_and_make_toas(
+        build_model, build_piecewise_model_with_two_pieces, "overlapping groups"
+    )
+    # m_piecewise.setup()
+    rs_value = pint.residuals.Residuals(
+        toas, m_piecewise, subtract_mean=False
+    ).resids_value
+    parameter_string = f"{param}_{convert_int_into_index(index)}"
+
+    output_param = []
+    m_piecewise_temp = add_offset_in_model_parameter(index, param, m_piecewise)
+
+    rs_temp = pint.residuals.Residuals(
+        toas, m_piecewise_temp, subtract_mean=False
+    ).resids_value
+    have_residuals_changed = np.invert(rs_temp == rs_value)
```

Comment:
you can use `rs_temp != rs_value`

Reply:
Talk about going the long way to get the same thing! Silly oversight on my part